ndcube/ndcollection.py

```diff
@@ -0,0 +1,261 @@
+                                for key, cube_item in zip(self, collection_items)]
+                    # Since item is not strings, no cube in collection is dropped.
+                    # Therefore the collection keys remain unchanged.
+                    new_keys = list(self.keys())
+
+            return self.__class__(list(zip(new_keys, new_data)), aligned_axes=new_aligned_axes,
+                                  meta=self.meta, sanitize_inputs=False)
+
+    def _generate_collection_getitems(self, item):
+        # There are 3 supported cases of the slice item: int, slice, tuple of ints and/or slices.
+        # Compile appropriate slice items for each cube in the collection and
+        # and drop any aligned axes that are sliced out.
+
+        # First, define empty lists of slice items to be applied to each cube in collection.
+        collection_items = [[slice(None)] * len(self[key].dimensions) for key in self]
+        # Define empty list to hold aligned axes dropped by the slicing.
+        drop_aligned_axes_indices = []
+
+        # Case 1: int
+        # First aligned axis is dropped.
+        if isinstance(item, int):
+            drop_aligned_axes_indices = [0]
+            # Insert item to each cube's slice item.
+            for i, key in enumerate(self):
+                collection_items[i][self.aligned_axes[key][0]] = item
+
+        # Case 2: slice
+        elif isinstance(item, slice):
+            # Insert item to each cube's slice item.
+            for i, key in enumerate(self):
+                collection_items[i][self.aligned_axes[key][0]] = item
+            # Note that slice interval's of 1 result in an axis of length 1.
+            # The axis is not dropped.
+
+        # Case 3: tuple of ints/slices
+        # Search sub-items within tuple for ints or 1-interval slices.
+        elif isinstance(item, tuple):
+            # Ensure item is not longer than number of aligned axes
+            if len(item) > self.n_aligned_axes:
+                raise IndexError("Too many indices")
+            for i, axis_item in enumerate(item):
+                if isinstance(axis_item, int):
+                    drop_aligned_axes_indices.append(i)
+                for j, key in enumerate(self):
+                    collection_items[j][self.aligned_axes[key][i]] = axis_item
+
+        else:
+            raise TypeError(f"Unsupported slicing type: {axis_item}")
+
+        # Use indices of dropped axes determine above to update aligned_axes
+        # by removing any that have been dropped.
+        drop_aligned_axes_indices = np.array(drop_aligned_axes_indices)
+        new_aligned_axes = collection_utils._update_aligned_axes(
+                drop_aligned_axes_indices, self.aligned_axes, self._first_key)
+
+        return collection_items, new_aligned_axes
+
+    def copy(self):
+        return copy.deepcopy(self)
+
+    def setdefault(self):
+        raise NotImplementedError("NDCollection does not support setdefault.")
```

Comment:
You also need to add a `__setitem__` which raises `NotImplementedError`.

Reply:
Argh!  We can't do this.  ```__setitem__``` is called under the hood by ```copy```.  So it causes ```NDCollection.copy()``` and copy.deepcopy(my_ndcollection)``` to error.